api/fees/fees.go

```diff
@@ -42,70 +45,111 @@ func calcPriorityFee(baseFee *big.Int, priorityPercentile int64) *big.Int {
 
 func New(repo *chain.Repository, bft bft.Committer, stater *state.Stater, config Config) *Fees {
 	return &Fees{
-		data:           newFeesData(repo, config.FixedCacheSize),
+		data:           newFeesData(repo, stater, config.FixedCacheSize),
 		bft:            bft,
-		stater:         stater,
 		minPriorityFee: calcPriorityFee(big.NewInt(thor.InitialBaseFee), int64(config.PriorityIncreasePercentage)),
 		config:         config,
 	}
 }
 
-func (f *Fees) validateGetFeesHistoryParams(req *http.Request) (uint32, *chain.BlockSummary, error) {
-	// Validate blockCount
+func (f *Fees) validateGetFeesHistoryParams(req *http.Request) (uint32, *chain.BlockSummary, []float64, error) {
+	blockCount, err := f.validateBlockCount(req)
+	if err != nil {
+		return 0, nil, nil, err
+	}
+
+	newestBlockSummary, err := f.validateNewestBlock(req, blockCount)
+	if err != nil {
+		return 0, nil, nil, err
+	}
+
+	rewardPercentiles, err := f.validateRewardPercentiles(req)
+	if err != nil {
+		return 0, nil, nil, err
+	}
+
+	return uint32(*blockCount), newestBlockSummary, rewardPercentiles, nil
+}
+
+func (f *Fees) validateBlockCount(req *http.Request) (*uint64, error) {
 	blockCountParam := req.URL.Query().Get("blockCount")
 	blockCount, err := strconv.ParseUint(blockCountParam, 10, 32)
 	if err != nil {
-		return 0, nil, utils.BadRequest(errors.WithMessage(err, "invalid blockCount, it should represent an integer"))
+		return nil, utils.BadRequest(errors.WithMessage(err, "invalid blockCount, it should represent an integer"))
 	}
 
 	if blockCount == 0 {
-		return 0, nil, utils.BadRequest(errors.New("invalid blockCount, it should not be 0"))
+		return nil, utils.BadRequest(errors.New("invalid blockCount, it should not be 0"))
 	}
 
-	// Validate newestBlock
+	return &blockCount, nil
+}
+
+func (f *Fees) validateNewestBlock(req *http.Request, blockCount *uint64) (*chain.BlockSummary, error) {
```

Comment:
If the only reason why the block count is passed by its pointer is we can change it, can we return it instead of changing it silently?

Reply:
Yeah, that was the reason. Modified and pushed, thanks.